source/utils/bitset.cpp

```diff
@@ -0,0 +1,84 @@
+#include "utils/bitset.hpp"
+
+#include <spdlog/fmt/fmt.h>
+
+#include "logger/logger.hpp"
+
+namespace sim {
+
+BitSet32::BitSet32() : m_data(0) {}
+
+BitSet32::BitSet32(std::uint32_t initial) : m_data(initial) {}
+
+bool BitSet32::set_bit(std::uint32_t pos, bool value) {
+    if (pos >= 32) {
```

Comment:
Please, use `sizeof(m_data)` instead of magic integer 32
```suggestion
    if (pos >= sizeof(m_data)) {
```